rewrite-java/src/main/java/org/openrewrite/java/cleanup/RemoveCallsToSystemGc.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2022 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.cleanup;
+
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.JavaType;
+import org.openrewrite.java.tree.TypeUtils;
+
+import java.time.Duration;
+import java.util.Collections;
+import java.util.Set;
+
+public class RemoveCallsToSystemGc extends Recipe {
+    @Override
+    public String getDisplayName() {
+        return "Remove calls to `System.gc()` method";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Removes calls to System.gc()";
+    }
+
+    @Override
+    public Set<String> getTags() {
+        return Collections.singleton("RSPEC-1215");
+    }
+
+    @Override
+    public Duration getEstimatedEffortPerOccurrence() {
+        return Duration.ofMinutes(1);
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return new NoGcVisitor();
+    }
+
+    private static class NoGcVisitor extends JavaIsoVisitor<ExecutionContext> {
+
+        @Override
+        public J.MethodInvocation visitMethodInvocation(J.MethodInvocation method, ExecutionContext context) {
+            J.MethodInvocation invocation = super.visitMethodInvocation(method, context);
+            if (invocation.getSelect() != null) {
+                final JavaType.FullyQualified fq = TypeUtils.asFullyQualified(invocation.getSelect().getType());
+                if (fq != null) {
+                    final Boolean isSystem = "java.lang.System".equals(fq.getFullyQualifiedName());
+                    final Boolean isGc = "gc".equals(method.getName().getSimpleName());
```

Comment:
Are you also familiar with the `org.openrewrite.java.MethodMatcher`?
Look for instance at [how this recipe uses MethodMatcher to match invocations](https://github.com/openrewrite/rewrite-migrate-java/blob/main/src/main/java/org/openrewrite/java/migrate/lang/StringFormatted.java#L35).

Reply:
See org.openrewrite.java.MethodMatcher for the exact matching pattern.

Reply:
Does it support matching against `void` return type? I tried that and failed, to be honest.

Reply:
I'd think it would; haven't got a working environment here right now, but I imagine this would be all you need:
```java
@Override
public TreeVisitor<?, ExecutionContext> getVisitor() {
    return new JavaIsoVisitor<ExecutionContext>() {
	@Override
	public J.MethodInvocation visitMethodInvocation(J.MethodInvocation method, ExecutionContext context) {
		J.MethodInvocation invocation = super.visitMethodInvocation(method, context);
		if (new MethodMatcher("java.lang.System gc()").matches(invocation)) {
			return null;
		}
		return invocation;
	}
    };
}
```

Reply:
Thanks. I wouldn't guess that :blush:. I thought such a string would match against methods which declare `java.lang.System` as the return type. Needless to say - your suggestion works

Reply:
Glad to help, and nice first addition. The matching patterns threw me off initially as well, [but it's documented here](https://github.com/openrewrite/rewrite/blob/main/rewrite-java/src/main/java/org/openrewrite/java/MethodMatcher.java#L44). Good to see you got it working!

Optionally you can extract `new MethodMatcher("java.lang.System gc()")` into a constant, and pass that into the `new UsesMethod<>(MethodMatcher)` constructor. That saves you from declaring `java.lang.System gc()` as String twice.

Reply:
Thank you.